### ISSUE TITLE
Network numbers and RTMP

### ISSUE THREAD

**Issue by cheesestraws** [sfiera/multitalk#1]
This issue brought to you by insomnia, the leading cause of hobbyist software engineering! </sponsor>

I think there's an issue with the handling of network numbers on the EtherTalk side (and perhaps TCP-side) here.  The extended networks, at any rate.  The problem is in this question: "what happens when you have an ethertalk node that has self-assigned an address not in defaultNet?"

According to Inside Appletalk (2nd Ed, p. 4-7), EtherTalk nodes should self-assign a network ID in the startup range $FF00 - $FFEE.  If the node chooses a startup address in $FF00, then the code here should work fine.  The problem comes if it picks a network ID elsewhere in the range.

A packet from this machine being sent through this bridge will reach the LocalTalk node perfectly happily, with a DDP extended header to say "this is on another network number".  But the LocalTalk node then has a problem: how does it get a packet back?  What should it put in its LLAP destination header?  Since it isn't on its current network, it will reach for its router - and of course, it hasn't got one.  So the packet will get dropped on the floor, because LocalTalk can't have more than one network number on an l2 segment.

So, since we can't make LocalTalk into an extended network, I think there are two options.  The first is to do BINAT (ew).  The second is to turn EtherTalk into a non-extended network, which is probably going to be easier.

To do this, I think the code needs to do two things:

1. Respond to ZIP GetNetInfo requests with a single network number
2. Occasionally send out RTMP Data packets with the network number in.

This way, the nodes on the EtherTalk will think that they are on a network with a single network number, nodes on the LocalTalk side will know what their network number _is_, and nobody will need to try to hit a router that doesn't exist.

Of course, for bonus points, one could try to play nicely, by:

1. On startup, wait a bit and see if we get any RTMP Data packets broadcast from a router upstream.  If those data packets contain a range of network numbers, throw an error and die; nothing we can do here.  
2. If, on the other hand, those contain a single network number, set the defaultNet to the number gleaned from them, and suppress generation of fake RTMP/ZIP.

To see whether this is actually the case / useful, just making defaultNet settable as a command line flag would be handy, then using an AppleTalk router to set the actual network number on the network to the same thing....

I hope this all makes sense.  It's been a long day.

**Comment by cheesestraws** [comment]
Actually, I'm not sure if this is right: is the LocalTalk driver aware of the EtherTalk startup range and considers it "local" anyway?  Wording in IA is unclear.  Wish this documentation were better.

**Comment by sfiera** [comment]
Yeah, I’d concluded that RTMP was going to be necessary. I see packets from the RTMP stub when a machine boots, so I was thinking that it would be enough to respond to those, but I hadn’t checked into ZIP yet. The idea of using a real router’s RTMP/ZIP packets as a first step is a good one. My holiday vacation starts Thursday, so maybe from then I’ll have a chance to look into this.
